File: CS/DXSample/ViewModels/TimeRegionViewModel.cs
using System;
using System.Windows.Media;
using DevExpress.Mvvm;

namespace DXSample.ViewModels {
    public class TimeRegionViewModel : ViewModelBase {
        public int Id {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
        public DateTime StartTime {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public DateTime EndTime {
            get { return GetValue<DateTime>(); }
            set { SetValue(value); }
        }
        public SolidColorBrush Brush {
            get { return GetValue<SolidColorBrush>(); }
            set { SetValue(value); }
        }
        public int ResourceId {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
    }
}

File: VB/DXSample/ViewModels/TimeRegionViewModel.vb
Imports System.Windows.Media
Imports DevExpress.Mvvm

Namespace DXSample.ViewModels

    Public Class TimeRegionViewModel
        Inherits ViewModelBase

        Public Property Id As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property

        Public Property StartTime As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property EndTime As Date
            Get
                Return GetValue(Of Date)()
            End Get

            Set(ByVal value As Date)
                SetValue(value)
            End Set
        End Property

        Public Property Brush As SolidColorBrush
            Get
                Return GetValue(Of SolidColorBrush)()
            End Get

            Set(ByVal value As SolidColorBrush)
                SetValue(value)
            End Set
        End Property

        Public Property ResourceId As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property
    End Class
End Namespace
